packages/swing-store/src/snapStore.js

```diff
@@ -203,70 +167,69 @@ export function makeSnapStore(
    */
-  async function saveSnapshot(vatID, snapPos, saveRaw) {
+  async function saveSnapshot(vatID, snapPos, snapshot) {
     const cleanup = [];
     return aggregateTryFinally(
       async () => {
-        // TODO: Refactor to use tmpFile rather than tmpName.
-        const tmpSnapPath = await ptmpName({ template: 'save-raw-XXXXXX.xss' });
-        cleanup.push(() => unlink(tmpSnapPath));
-        const { duration: rawSaveSeconds } = await measureSeconds(async () =>
-          saveRaw(tmpSnapPath),
-        );
-        const { size: uncompressedSize } = await stat(tmpSnapPath);
-
-        // Perform operations that read snapshot data in parallel.
-        // We still serialize the stat and opening of tmpSnapPath
-        // and creation of tmpGzPath for readability, but we could
-        // parallelize those as well if the cost is significant.
-        const snapReader = createReadStream(tmpSnapPath);
-        cleanup.push(() => {
-          snapReader.destroy();
-        });
-
-        await fsStreamReady(snapReader);
-
         const hashStream = createHash('sha256');
         const gzip = createGzip();
         let compressedSize = 0;
+        let uncompressedSize = 0;
 
-        const { result: hash, duration: compressSeconds } =
+        const { duration: compressSeconds, result: compressedSnapshot } =
           await measureSeconds(async () => {
+            const snapReader = Readable.from(snapshot);
+            cleanup.push(
+              () =>
+                new Promise((resolve, reject) =>
+                  snapReader.destroy(
+                    null,
+                    // @ts-expect-error incorrect types
+                    err => (err ? reject(err) : resolve()),
+                  ),
+                ),
+            );
+
+            snapReader.on('data', chunk => {
+              uncompressedSize += chunk.length;
+            });
             snapReader.pipe(hashStream);
-            const compressedSnapshot = await buffer(snapReader.pipe(gzip));
+            const compressedSnapshotData = await buffer(snapReader.pipe(gzip));
             await finished(snapReader);
-
-            const h = hashStream.digest('hex');
-            ensureTxn();
-            stopUsingLastSnapshot(vatID);
-            compressedSize = compressedSnapshot.length;
-            sqlSaveSnapshot.run(
-              vatID,
-              snapPos,
-              1,
-              h,
-              uncompressedSize,
-              compressedSize,
-              compressedSnapshot,
-            );
-            const rec = snapshotRec(vatID, snapPos, h, 1);
-            const exportKey = snapshotMetadataKey(rec);
-            noteExport(exportKey, JSON.stringify(rec));
-            noteExport(
-              currentSnapshotMetadataKey(rec),
-              snapshotArtifactName(rec),
-            );
-            return h;
+            return compressedSnapshotData;
           });
+        const hash = hashStream.digest('hex');
```

Comment:
Let's call this `snapshotID`, that's the name I started using recently in the new transcript entries.

Although.. the DB schema might use `hash`. If so, I'd be in favor of changing the schema.